src/ConnectedMode/Migration/MSBuildFileCleaner.cs

```diff
@@ -37,7 +37,7 @@ public MSBuildFileCleaner(ILogger logger)
             this.logger = logger;
         }
 
-        public Task CleanAsync(string filePath, LegacySettings legacySettings, CancellationToken token)
+        public Task CleanAsync(string filePath, LegacySettings legacySettings, IVsAwareFileSystem fileSystem,  CancellationToken token)
```

Comment:
What if instead of mutable IFileToClean we made it Immutable and returned ICleanFile for files that had some cleaning done and skip the other files?

Reply:
So:

```
public Task<ICleaned> CleanAsync(IFileToClean file, LegacySettings legacySettings, CancellationToken token)
```

What would be returned if there were no changes? `null`?

Reply:
Discussed offline: the cleaner doesn't need any additional information (e.g. file type, file name) - it just needs the text i.e. it won't differentiate between cleaning project files and rulesets.

It also doesn't need to return anything except the modified text i.e. no additional error messages -> it can return just a string (if it turns out we need to return extra info we can change the return type to a custom object).

So the interface becomes:
```
    internal interface IFileCleaner
    {
        Task<string> CleanAsync(string text, LegacySettings legacySettings, CancellationToken token);
    }
```